test/powershell/Modules/Microsoft.PowerShell.Security/ConstrainedLanguageRestriction.Tests.ps1

```diff
@@ -0,0 +1,890 @@
+            }
+        }
+
+        AfterEach {
+            Remove-SystemLockDown
+        }
+    }
+
+    Describe "Mkdir command on function drive in constrained language" -Tags 'Feature' {
+
+        It "Verifies mkdir on function drive is not allowed in constrained language mode" {
+
+            $ExecutionContext.SessionState.LanguageMode = "ConstrainedLanguage"
+
+            try
+            {
+                mkdir function:\SomeEvilFunction -Value SomeBadScriptBlock -ErrorAction Stop
+                throw "No Exception!"
+            }
+            catch
+            {
+                $exception = $_
+            }
+
+            Invoke-LanguageModeTestingSupportCmdlet -EnableFullLanguage
+
+            $exception.FullyQualifiedErrorId | Should Match "NotSupported"
+        }
+
+        AfterAll {
+            Invoke-LanguageModeTestingSupportCmdlet -EnableFullLanguage
+        }
+    }
+
+    Describe "Script debugging in constrained language" -Tags 'Feature' {
+
+        It "Verifies that a debugging breakpoint cannot be set in constrained language and no system lockdown" {
+
+            $ExecutionContext.SessionState.LanguageMode = "ConstrainedLanguage"
+
+            try
+            {
+                function MyDebuggerFunction {}
+                Set-PSBreakpoint -Command MyDebuggerFunction
+                throw "No Exception!"
+            }
+            catch
+            {
+                $exception = $_
+            }
+
+            Invoke-LanguageModeTestingSupportCmdlet -EnableFullLanguage
+
+            $exception.FullyQualifiedErrorId | Should Match "CannotSetBreakpointInconsistentLanguageMode"
+        }
+        
+        # TODO
+        # Test is pending because PowerShell Core currently stubs out Windows system lockdown APIs
+        It "Verifies that a debugging breakpoint can be set in constrained language with system lockdown" -Pending {
+
+            Set-SystemLockDown
+
+            function MyDebuggerFunction2 {}
+            $Global:DebuggingOk = $null
+            $null = Set-PSBreakpoint -Command MyDebuggerFunction2 -Action { $Global:DebuggingOk = "DebuggingOk" }
+            MyDebuggerFunction2
+
+            Remove-SystemLockDown
+
+            $Global:DebuggingOk | Should Be "DebuggingOk"
+        }
+
+        # TODO
+        # Test is pending because PowerShell Core currently stubs out Windows system lockdown APIs
+        It "Verifies that debugger commands do not run in full language mode when system is locked down" -Pending {
+
+            $Error.Clear()
+            Set-SystemLockDown
+
+            function MyDebuggerFunction3 {}
+            $null = Set-PSBreakpoint -Command MyDebuggerFunction3 -Action { $Global:dbgResult = [object]::Equals("A", "B") }
+
+            MyDebuggerFunction3
+            $expectedError = $Error[0]
+
+            Remove-SystemLockDown
+
+            $expectedError.FullyQualifiedErrorId | Should Match "MethodInvocationNotSupportedInConstrainedLanguage"

```

Comment:
i would expect this could be done as `try { } catch { } finally { }` if `-ErrorAction Stop` was used

Reply:
Ok.